zerver/lib/streams.py

```diff
@@ -423,7 +462,23 @@ def check_for_exactly_one_stream_arg(stream_id: int | None, stream: str | None)
         raise IncompatibleParametersError(["stream_id", "stream"])
 
 
-def check_stream_access_for_delete_or_update(
+def channel_admin_has_content_access(
+    user_profile: UserProfile, stream: Stream, sub: Subscription | None = None
+) -> bool:
+    if not stream.invite_only or sub is not None:
+        return True
+
+    user_recursive_group_ids = set(
+        get_recursive_membership_groups(user_profile).values_list("id", flat=True)
+    )
```

Comment:
We could have not used this and instead used `is_user_in_group`, but this approach has the same queries as that and with this approach, we can check for `can_subscribe_group` in future without any additional query.